app/models/attachment_data.rb

```diff
@@ -117,7 +117,11 @@ def deleted?
   end
 
   def draft?
-    !significant_attachable.publicly_visible?
+    if unpublished?
+      true
```

Comment:
I don't think this is correct.

This would mark attachments which existed when the document was unpublished as draft, which then would send people to the draft-assets host when they tried to access them, rather than redirecting to the "sorry, this page was unpublished" page.